backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/impexp/CampaignPriceImportTableSqlUpdater.java

```diff
@@ -0,0 +1,222 @@
+				.append("WHERE C_Country_ID IS NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "='N'")
+				.append(selection.toSqlWhereClause("i"));
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+
+		sql = new StringBuilder("UPDATE ")
+				.append(targetTableName)
+				.append(" SET " + COLUMNNAME_I_IsImported + "='E', " + COLUMNNAME_I_ErrorMsg + "=" + COLUMNNAME_I_ErrorMsg + "||'ERR=Invalid Country,' ")
+				.append("WHERE C_Country_ID IS NULL AND CountryCode IS NOT NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "<>'Y'")
+				.append(selection.toSqlWhereClause());
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+	}
+
+	private static void dbUpdateCurrency(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET C_Currency_ID=(SELECT C_Currency_ID FROM C_Currency c")
+				.append(" WHERE i.ISO_Code=c.ISO_Code  LIMIT 1) ")
+				.append("WHERE C_Currency_ID IS NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "='N'")
+				.append(selection.toSqlWhereClause("i"));
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+
+		sql = new StringBuilder("UPDATE ")
+				.append(targetTableName)
+				.append(" SET " + COLUMNNAME_I_IsImported + "='E', " + COLUMNNAME_I_ErrorMsg + "=" + COLUMNNAME_I_ErrorMsg + "||'ERR=Invalid Currency,' ")
+				.append("WHERE C_Currency_ID IS NULL AND ISO_Code IS NOT NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "<>'Y'")
+				.append(selection.toSqlWhereClause());
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+	}
+
+	private static void dbUpdateUom(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET C_UOM_ID=(SELECT C_UOM_ID FROM C_UOM u")
+				.append(" WHERE i.UOM=u.name AND u.AD_Client_ID in (i.AD_Client_ID, 0) LIMIT 1) ")
+				.append("WHERE C_UOM_ID IS NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "='N'")
+				.append(selection.toSqlWhereClause("i"));
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+
+		sql = new StringBuilder("UPDATE ")
+				.append(targetTableName)
+				.append(" SET " + COLUMNNAME_I_IsImported + "='E', " + COLUMNNAME_I_ErrorMsg + "=" + COLUMNNAME_I_ErrorMsg + "||'ERR=Invalid UOM,' ")
+				.append("WHERE C_UOM_ID IS NULL AND UOM IS NOT NULL")
+				.append(" AND " + COLUMNNAME_I_IsImported + "<>'Y'")
+				.append(selection.toSqlWhereClause());
+		DB.executeUpdateEx(sql.toString(), ITrx.TRXNAME_ThreadInherited);
+	}
+
+	private static void dbUpdateTaxCategory(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET C_TaxCategory_ID=(SELECT C_TaxCategory_ID FROM C_TaxCategory tc")
+				.append(" WHERE i.C_TaxCategory_Name=tc.Name AND tc.AD_Client_ID in (i.AD_Client_ID, 0) LIMIT 1) ")
```

Comment:
the taxcategory-name is unique among AD_Orgs but can be duplicated among AD_Client_IDs?
(also: please see the comment about `dbUpdateProduct()`)

Reply:
![image](https://user-images.githubusercontent.com/60300682/165275477-b152f3eb-2198-4069-b537-be7cc2819432.png)